helpers.go

```diff
@@ -135,6 +135,18 @@ func deserialize(b []byte) (proto.Message, error) {
 	return a.UnmarshalNew()
 }
 
+func serializePayload(m proto.Message) ([]byte, *anypb.Any, error) {
+	b, err := proto.Marshal(m)
+	if err != nil {
+		return nil, nil, err
+	}
+	a := &anypb.Any{
+		TypeUrl: string(m.ProtoReflect().Descriptor().FullName()),
```

Comment:
`anypb.New(m)`, this will break

Reply:
it doesn't. the url prefix is completely superficial. iirc the resolver discards it when finding the type

Reply:
yep https://github.com/protocolbuffers/protobuf-go/blob/6875c3d7242d1a3db910ce8a504f124cb840c23a/reflect/protoregistry/registry.go#L646-L649

Reply:
playing with fire

```
// MessageIs reports whether the underlying message is of the same type as m.
func (x *Any) MessageIs(m proto.Message) bool {
	if m == nil {
		return false
	}
	url := x.GetTypeUrl()
	name := string(m.ProtoReflect().Descriptor().FullName())
	if !strings.HasSuffix(url, name) {
		return false
	}
	return len(url) == len(name) || url[len(url)-len(name)-1] == '/'
}
```

Reply:
we don't use this. i'm confident it's safe but swapped